nodebuilder/core/flags.go

```diff
@@ -50,11 +50,18 @@ func ParseFlags(
 		return nil
 	}
 
-	rpc := cmd.Flag(coreRPCFlag).Value.String()
-	grpc := cmd.Flag(coreGRPCFlag).Value.String()
+	// in the case that the RPC port is not set, use the default port
+	// or in the case a value has been set via flag,
+	// set that as highest precedence
+	if cfg.RPCPort == "" || cmd.Flag(coreRPCFlag).Changed {
+		rpc := cmd.Flag(coreRPCFlag).Value.String()
+		cfg.RPCPort = rpc
+	}
+	if cfg.GRPCPort == "" || cmd.Flag(coreGRPCFlag).Changed {
```

Comment:
This allows an invalid configuration to be put in place with empty fields. WDYT about prohibiting that and failing the user with Validate? We could only detect that the flag has been changed and alter the config via `cmd.Flag(coreGRPCFlag).Change`

Reply:
This actually prevents an empty config being accidentally preserved after ParseFlags if the default flag is not changed and an empty config  is passed in. We never get here because we always (I think) get NodeConfig with defaults. I only encountered from writing tests, feels like its worth a check

agree with adding empty checks to cfg.Validate() ...I'll add

Reply:
> This actually prevents an empty config being accidentally preserved after ParseFlags if the default flag is not changed and an empty config is passed in.

That's true. My suggestion is to fail if an empty config fields were passed, as opposed to populating them. So we only override fields if flags change, allowing the empty fields case to fail during further validation in ParseFlags